spec/datadog/appsec/component_spec.rb

```diff
@@ -0,0 +1,45 @@
+# typed: ignore
+
+require 'datadog/appsec/spec_helper'
+require 'datadog/appsec/component'
+
+RSpec.describe Datadog::AppSec::Component do
+  describe '.build_appsec_component' do
+    context 'when appsec is enabled' do
+      it 'returns a Datadog::AppSec::Component instance' do
+        settings = Datadog::AppSec::Configuration::Settings.new.merge(
+          Datadog::AppSec::Configuration::DSL.new.tap do |it|
+            it.enabled = true
```

Comment:
Minor: this `it` is confusing, as it is not the same `it` as RSpec's DSL, nor the `subject` of the test case. I feel this would have been more readable:

```
          Datadog::AppSec::Configuration::DSL.new.tap do |appsec|
            appsec.enabled = true
```